### ISSUE TITLE
Preserving radio settings changed/broken in pre-release version?

### ISSUE THREAD

**Issue by Snak3Doc** [michail-nikolaev/task-force-arma-3-radio#1303]
**Arma 3 Version:** `1.74` (stable)
**CBA Version:** `3.4.0.170627` (stable)
**Task Force Radio Version:** `1.-1.0.247` (Pre Release)

**Mods:**
- @CBA_A3
- @ACE
- @task_force_radio
- @ILBE radios
- @backpackonchest by DerZade
- HALO Script
- Other Non-related mods

**Description:**
- When doing actions like extend/retract antenna on ILBE radios, swapping backpack with @backpackonchest, or using a HALO script, the players backpack radio settings are no longer retained using the above version of TFR. I was wondering if the code for this had changed in the Pre-release version as DerZade just confirmed to me that it was working for him on stable TFR version, if it has changed could someone point me in the direction of those changes please?

**Steps to reproduce:**
- Use any of the above mods that make use of preserving radio settings function ie extend/retract antenna and your radio frequency will not be preserved 

**Where did the issue occur?**
- Dedicated Windows x64

**Comment by dedmen** [comment]
Can you show me that Halo script? Whatever moves the backpack apparently doesn't keep the variables on it. Nothing on that changed from 0.9.12. Variable name and location is still the same.

**Comment by Snak3Doc** [comment]
Yeah I had a look through for differences myself and couldn't find any, theirs the HALO script I'm using https://dropfile.to/0Aq6Nte

**Comment by dedmen** [comment]
Link down.. In europe atleast.

**Comment by Snak3Doc** [comment]
Odd fine in aus try that https://drive.google.com/file/d/0BxPqbaPPBxSJVTdWcElVaWJWcEU/view?usp=sharing

**Comment by dedmen** [comment]
Yeah. That Mod doesn't preserve the Variables on the Backpack. Thus getting reset.
Same would happen in 0.9.x. Except if you set default frequencies and didn't change them.. Then ofcause they would stay the same.

**Comment by Snak3Doc** [comment]
Ahh, right I thought it had that code in it, my bad man.
This is the mod made by DerZade http://steamcommunity.com/sharedfiles/filedetails/?id=820924072&searchtext=backpackonchest
It's been updated recently and the author confirmed to me the preserving of channels was working for him on 0.9.x. stable when swapping backpacks from chest to back.

**Comment by dedmen** [comment]
That's because setLRSettings now takes array and settings instead of 2 values and settings.
I'll make a PR. It's a very easy fix.

**Comment by dedmen** [comment]
https://github.com/DerZade/BackpackOnChest/pull/24

**Comment by Snak3Doc** [comment]
Awesome, many thanks man. I'm just going through the changes now, I should be able to use that PR to add this to the HALO script and patch ILBE LR backpacks mod as well correct?

**Comment by dedmen** [comment]
Yes you could. Does ILBE have a github repo?

**Comment by dedmen** [comment]
I will turn this into a enhancement issue.
I can detect if someone calls setLrSettings with 3 arguments instead of the expected 2.
If that happens I can use the old way of parsing the parameters. That way nothing will get broken.

**Comment by Snak3Doc** [comment]
That sounds like an easier option when it comes to abandoned mods like ILBE, either way, i believe this is the original repo, https://github.com/Raspu86/tfw_ilbe

**Comment by Snak3Doc** [comment]
Just made some kind of PR to fix missing "then" condition on line 25 and missing semicolon on line 29, still learning how GH works not sure if it went to you or DerZade

**Comment by dedmen** [comment]
Went to me. But It's fine.

**Comment by dedmen** [comment]
https://github.com/michail-nikolaev/task-force-arma-3-radio/issues/1303#issuecomment-325312254
